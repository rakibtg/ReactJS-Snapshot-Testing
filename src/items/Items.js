import React from 'react'

const Items = ({ items = [] }) => {

    const [listOfItems, itemsUpdator] = React.useState(items)

    if (listOfItems.length <= 0) {
        return <div>No items found</div>
    } else {
        return <div>
            <ol>
                {
                    listOfItems.map((item, index) => {
                        return <li key={index}>{item}</li>
                    })
                }
            </ol>
        </div>
    }
}

export default Items;