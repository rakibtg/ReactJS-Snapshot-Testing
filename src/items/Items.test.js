import React from 'react';
import renderer from 'react-test-renderer';

import Items from './Items';

it ("renders when there are no items", () => {
    const tree = renderer.create(
        <Items />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it ("renders when there is one item", () => {
    const tree = renderer.create(
        <Items items={["Eath"]} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
})

it ("renders when there are multiple items", () => {
    const tree = renderer.create(
        <Items items={['Sun', 'Earth', 'Moon', 'Saturn']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
})