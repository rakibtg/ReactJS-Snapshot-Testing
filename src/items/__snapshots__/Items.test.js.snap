// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are multiple items 1`] = `
<div>
  <ol>
    <li>
      Sun
    </li>
    <li>
      Earth
    </li>
    <li>
      Moon
    </li>
    <li>
      Saturn
    </li>
  </ol>
</div>
`;

exports[`renders when there are no items 1`] = `
<div>
  No items found
</div>
`;

exports[`renders when there is one item 1`] = `
<div>
  <ol>
    <li>
      Eath
    </li>
  </ol>
</div>
`;
